### ISSUE TITLE
Discord admin chat text is being improperly sent globally.

### ISSUE THREAD

**Issue by mibby** [kangarko/ChatControl-Pro#1546]
Paper dev 87 (Spigot 1.15.2)
ChatControl Pro v8.5.7
DiscordSRV v1.18.4

(Edit) Also tested DiscordSRV-Build-1.18.5-SNAPSHOT-1580563155 for latest development branch changes. Still broken.

For some reason, typing in my discord staff channel is sending messages globally instead of to ChatControl's admin chat channel. This use to work fine for me but seems to have broke within some recent update or change to ChatControl. Do you have any ideas on how to fix this? :(

* Typing in in-game global chat, it is properly sending the message to ``#in-game`` channel on discord.
* Typing in in-game admin chat, it is properly sending the message to ``#staff-in-game`` channel on discord.
* Typing in discord ``#in-game`` channel, it is properly sending the message to in-game global chat.
* Typing in discord ``#staff-in-game`` channel, it **improperly** sends the message to in-game global chat _instead_ of in-game admin chat.

DiscordSRV config.yml
```
# Channel links from game to Discord
# syntax is Channels: {"in-game channel name": "numerical channel ID from Discord", "another in-game channel name": "another numerical channel ID from Discord"}
# The first channel pair specified in this config option will be the "main" channel, used for sending player joins/quits/deaths/achievements/etc
#
Channels: {"in-game": "280012581983354881", "staff-in-game": "494635905013907457"}
```

ChatControl settings.yml
```
Admin_Chat:
  Bungee: true
  Log_To_Console: true
  Write_To_File: true
  Discord_Channel: "staff-in-game"
  Shortcut: none
Bungee_Chat:
  Enabled_By_Default: true
  Log_To_Console: true
  Write_To_File: false
  Discord_Channel: "in-game"
  Shortcut: '!'
Listener_Priority:
  Checker: MONITOR
  Formatter: MONITOR
  Signs: HIGHEST
BungeeCord:
  Enabled: true
  Prefix: ''
```

ChatControl formatting.yml
```
Integration:
  Discord_Channel: "in-game"
```

**Comment by lackoSK** [comment]
Hello,

When did it worked correctly? To which version have you updated?

Regards,
lackoSK

**Comment by mibby** [comment]
Sorry, I do not recall exactly when it began to break. I just know it use to work fine in 1.14.4 builds and I think the 8.5.0-BETA build for 1.15? I tend to update every time a new version of ChatControl comes out and it's only recently I've been hearing complaints about it not working.

**Comment by lackoSK** [comment]
Hello,

Like, we haven't done any changes to Discord integration. We can try to update it's API and see what happens. 

Regards,
lackoSK

**Comment by mibby** [comment]
Sure, let me know if you need me to test any special builds. As I cannot find any fault with how it is currently setup.

**Comment by mibby** [comment]
This is still an issue as of ChatControl Pro v8.6.4 and DiscordSRV v1.19.0-SNAPSHOT. Typing in my staff channel on discord sends the message to public chat instead of ChatControl's admin chat.

**Comment by lackoSK** [comment]
We have updated our repo. Please wait until the new release.

**Comment by lackoSK** [comment]
Out of my testing, it worked fine..

I will close it here. If the issue will continue, a message under this issue.

**Comment by mibby** [comment]
@lackoSK Doing some testing again, this is still not working for me with ChatControlPro v8.6.6 and DiscordSRV v1.19.0-SNAPSHOT (build 1584093078 jar).

DiscordSRV Config
```
# Channel links from game to Discord
# syntax is Channels: {"in-game channel name": "numerical channel ID from Discord", "another in-game channel name": "another numerical channel ID from Discord"}
# The first channel pair specified in this config option will be the "main" channel, used for sending player joins/quits/deaths/achievements/etc
#
Channels: {"global": "280012581983354881", "staff-chat": "494635905013907457"}
```

ChatControl Settings
```
Admin_Chat:
  Bungee: true
  Log_To_Console: true
  Write_To_File: true
  Discord_Channel: 'staff-chat'
  Shortcut: none
Bungee_Chat:
  Enabled_By_Default: true
  Log_To_Console: true
  Write_To_File: false
  Discord_Channel: 'global'
  Shortcut: none
```

Once again, typing on discord in the staff only channel sends the message to global chat, not to ChatControl's admin chat. 

Installing the plugin DiscordSRV-Staff-Chat, their staff chat both ways (discord & in-game) sends fine privately. This issue is with ChatControl. :(

**Comment by mibby** [comment]
@lackoSK @kangarko Hello? Is this not something that is going to be fixed? :( It's a bit silly to have the discord staff chat leak into ChatControl's global public chat instead of being privately sent to ChatControl's admin chat only.

**Comment by TheReverend403** [comment]
I can vouch that this is still happening.

**Comment by mibby** [comment]
Yep, going to bump this issue again. 

@lackoSK Please.

**Comment by kangarko** [comment]
We are rewriting our chat system, I am keeping an eye on this and should be out in the next 7-14 days!

**Comment by mibby** [comment]
For what it's worth, this issue seems fixed with v9.0.0. Staff chat no longer forwards chat to both discord channels.
